Prose/ProseLib/Sources/ConversationFeature/Chat/Chat.swift

```diff
@@ -281,6 +319,34 @@ struct ChatView: NSViewRepresentable {
     #endif
   }
 
+  func showReactionPicker(_ pickerState: MessageReactionPickerState, on webView: ChatWebView) {
+    logger.trace("Showing reaction picker…")
+
+    #if os(macOS)
+      let picker: ReactionPickerView = webView.reactionPicker(
+        store: self.store.scope(state: { _ in pickerState }),
+        action: CasePath(ChatAction.reactionPicker),
+        dismiss: .reactionPickerDismissed
+      )
+
+      picker.setFrameOrigin(pickerState.origin)
+
+      if picker.superview == nil {
+        webView.addSubview(picker)
+      }
+    #else
+      #warning("Show a reaction picker")
+    #endif
+  }
+
+  func hideReactionPicker(from webView: ChatWebView) {
+    logger.trace("Hiding reaction picker…")
+
+    #if os(macOS)
+      webView._reactionPicker?.removeFromSuperview()
```

Comment:
That's a bit weird. On the one hand you have a method on the ChatWebView which creates the reaction picker instance on the other hand you reach for the (private? - hence "_") property directly.

Instead of creating a subclass of the WKWebView for the sole purpose of storing a pointer to a view, I'd capture the view in the `sink` closure in `makeNSView` and either move the code of `showReactionPicker` and `hideReactionPicker` in there or pass it along to these methods.

Also starting property names with an underscore would be a bad precedent.

Reply:
I'll change this now

Reply:
Better, yes

Reply:
I increased the timeout, and hopefully the tests will always pass.